pkg/controller/job/job_controller.go

```diff
@@ -1023,7 +1023,7 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 		if podFinished || podTerminating || job.DeletionTimestamp != nil {
 			podsToRemoveFinalizer = append(podsToRemoveFinalizer, pod)
 		}
-		if pod.Status.Phase == v1.PodSucceeded {
+		if pod.Status.Phase == v1.PodSucceeded && !uncounted.failed.Has(string(pod.UID)) {
```

Comment:
regarding the comment above at line 1018, are we counting them as failed because we may not get another job sync to remove their finalizers?

Reply:
Correct, that is covered in line 1020.

Reply:
it is nit, but I don't think it is covered in that comment if the sole purpose is to "trigger another sync to remove the finalizers".